kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkConnectorConfig.java

```diff
@@ -320,6 +341,33 @@ public String toString() {
     }
   }
 
+  private static class CannedAclValidator implements ConfigDef.Validator {
+    private static String allowedValues;
+
+    static {
+      List<String> aclStrings = new ArrayList<>();
+      for (CannedAccessControlList acl: CannedAccessControlList.values()) {
+        aclStrings.add(acl.name());
+      }
+      allowedValues = Utils.join(aclStrings, ", ");
+    }
+
+    @Override
+    public void ensureValid(String name, Object cannedAcl) {
+      String aclStr = ((String) cannedAcl).trim();
+      try {
+        CannedAccessControlList.valueOf(aclStr);
```

Comment:
we need to compare against the actual values not the enum names. We'll need to reuse the `Map<String, CannedAccessControlList>` that we'll need for `getCannedAcl ` or a `Set<String>` to store the allowed values.